File: Domain/Entities/Prescription.cs
// Domain/Interfaces/Prescription.cs
namespace Domain.Entities
{
  public class Prescription
    {
        public Guid PrescriptionId { get; set; }
        public Guid PatientId { get; set; }
        public string MedicationName { get; set; }
        public string Dosage { get; set; }
        public string Frequency { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public string Notes { get; set; }

    // Navigation property
    public Patient Patient { get; set; }
  }
}

File: health-front/src/app/models/prescription.ts
export interface Prescription {
    prescriptionId: string;
    patientId: string;
    medicationName: string;
    dosage: string;
    frequency: string;
    startDate: string;
    endDate: string;
    notes: string;
}

export interface PrescriptionResponse extends Prescription {
    patientName?: string;
    doctorName?: string;
}

export interface PaginatedPrescriptionResponse {
    items: Prescription[];
    totalCount: number;
    page: number;
    pageSize: number;
}